src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,623 @@
+
+    /**
+     * Setter to specify check name pattern to suppress. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from target file.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public final void setCheckPattern(String pattern) {
+        checkPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check violation message pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setMessagePattern(String pattern) {
+        messagePattern = pattern;
+    }
+
+    /**
+     * Setter to specify check ID pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setIdPattern(String pattern) {
+        idPattern = pattern;
+    }
+
+    /**
+     * Setter to specify negative/zero/positive value that defines the number
+     * of lines preceding/at/following the suppressing nearby text. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from target file.
+     *
+     * @param format a {@code String} value.
+     */
+    public final void setLineRange(String format) {
+        lineRange = format;
+    }
+
+    @Override
+    public boolean accept(AuditEvent event) {
+        boolean accepted = true;
+
+        if (event.getViolation() != null) {
+            final String eventFileTextAbsolutePath = event.getFileName();
+
+            if (!currentFileAbsolutePath.equals(eventFileTextAbsolutePath)) {
+                final Optional<FileText> currentFileText = getFileText(eventFileTextAbsolutePath);
+
+                if (currentFileText.isPresent()) {
+                    final FileText fileText = currentFileText.get();
+                    currentFileAbsolutePath = fileText.getFile().getAbsolutePath();
+                    collectSuppressions(fileText);
+                }
+            }
+
+            final Optional<Suppression> nearestSuppression =
+                    getNearestSuppression(suppressions, event);
+            accepted = nearestSuppression.isEmpty();
+        }
+        return accepted;
+    }
+
+    @Override
+    protected void finishLocalSetup() {
+        // No code by default
+    }
+
+    /**
+     * Returns {@link FileText} instance created based on the given file name.
+     *
+     * @param fileName the name of the file.
+     * @return {@link FileText} instance.
+     * @throws IllegalStateException if the file could not be read.
+     */
+    private static Optional<FileText> getFileText(String fileName) {
+        final File file = new File(fileName);
+        FileText result = null;
+
+        if (!file.isDirectory()) {
+            try {
+                result = new FileText(file, StandardCharsets.UTF_8.name());
+            }
+            catch (IOException ex) {
+                throw new IllegalStateException("Cannot read source file: " + fileName, ex);
```

Comment:
@romani I think all our classes should allow us to throw `CheckstyleException`.

Reply:
yes, it should be allowed to use this Exception in whole our code

Reply:
We can't, because it is a checked exception and our `Filter`'s `accept` method does not declare it.

Reply:
lets move this to separate issue, better to merge this long lasting PR sooner.